odoo/addons/base/tests/test_res_users.py

```diff
@@ -534,3 +534,27 @@ def test_superuser(self):
         self.assertFalse(user.active)
         with self.assertRaises(UserError):
             user.write({'active': True})
+
+
+class TestUsersIdentitycheck(HttpCase):
+
+    @classmethod
+    def setUpClass(self):
+        super().setUpClass()
+        self.user_internal = self.env['res.users'].create({
+            'name': 'Internal',
+            'login': 'user_internal',
+            'password': 'admin',
+            'groups_id': [self.env.ref('base.group_user').id],
+        })
+
+    @users('user_internal')
+    def test_revoke_all_devices(self):
+        """
+        Test to check the revoke all devices by changing the current password as a new password
+        """
+        user_identity_check = self.env['res.users.identitycheck'].create({'request': True, 'password': 'admin'})
+        self.assertEqual(user_identity_check.password, 'admin')
+        user_identity_check.revoke_all_devices()
+        # After revoking all the devices user_identity_check data will be unlinked
+        self.assertFalse(user_identity_check.exists())
```

Comment:
You test that you fix the bug but you do not really test the feature itself, which is invalidationg all the sessions of the user.

As there is no test testing the feature, it's a good opportunity to do it.

In the end, applying the above comment + testing the feature, here is what I did:

```diff
diff --git a/odoo/addons/base/tests/test_res_users.py b/odoo/addons/base/tests/test_res_users.py
index fe936e071e90..89d67ec83765 100644
--- a/odoo/addons/base/tests/test_res_users.py
+++ b/odoo/addons/base/tests/test_res_users.py
@@ -7,6 +7,7 @@ from unittest.mock import patch
 from odoo import SUPERUSER_ID
 from odoo.addons.base.models.res_users import is_selection_groups, get_selection_groups, name_selection_groups
 from odoo.exceptions import UserError
+from odoo.service import security
 from odoo.tests.common import Form, TransactionCase, new_test_user, tagged, HttpCase, users
 from odoo.tools import mute_logger

@@ -536,25 +537,35 @@ class TestUsersTweaks(TransactionCase):
             user.write({'active': True})


+@tagged('post_install', '-at_install')
 class TestUsersIdentitycheck(HttpCase):

-    @classmethod
-    def setUpClass(self):
-        super().setUpClass()
-        self.user_internal = self.env['res.users'].create({
-            'name': 'Internal',
-            'login': 'user_internal',
-            'password': 'admin',
-            'groups_id': [self.env.ref('base.group_user').id],
-        })
-
-    @users('user_internal')
+    @users('admin')
     def test_revoke_all_devices(self):
         """
         Test to check the revoke all devices by changing the current password as a new password
         """
-        user_identity_check = self.env['res.users.identitycheck'].create({'request': True, 'password': 'admin'})
-        self.assertEqual(user_identity_check.password, 'admin')
+        # Create a session
+        session = self.authenticate('admin', 'admin')
+        # Test the session is valid
+        self.assertTrue(security.check_session(session, self.env))
+        # Valid session -> not redirected from /web to /web/login
+        self.assertTrue(self.url_open('/web').url.endswith('/web'))
+
+        # The user clicks the button logout from all devices from his profile
+        action = self.env.user.action_revoke_all_devices()
+        # The form of the wizard opens with a new record
+        form = Form(self.env[action['res_model']].create({}), action['view_id'])
+        # The user fills his password
+        form.password = 'admin'
+        # The user clicks the button "Log out from all devices", which triggers a save then a call to the button method
+        user_identity_check = form.save()
         user_identity_check.revoke_all_devices()
-        # After revoking all the devices user_identity_check data will be unlinked
+
+        # Test the session is no longer valid
+        self.assertFalse(security.check_session(session, self.env))
+        # Invalid session -> redirected from /web to /web/login
+        self.assertTrue(self.url_open('/web').url.endswith('/web/login'))
+
+        # In addition, the wizard must have been deleted
         self.assertFalse(user_identity_check.exists())
```